skt/executable.py

```diff
@@ -341,15 +341,20 @@ def cmd_build(args):
     # Gather additional details about the build and save them to the state
     # file.
     kernel_arch = builder.build_arch
-    cross_compiler_prefix = builder.cross_compiler_prefix
     make_opts = builder.assemble_make_options()
     state = {
         'kernel_arch': kernel_arch,
-        'cross_compiler_prefix': cross_compiler_prefix,
         'make_opts': make_opts
     }
     update_state(args['rc'], state)
 
+    # Write the cross compiler prefix to the state file only if the
+    # environment variable is set:
+    cross_compiler_prefix = builder.cross_compiler_prefix
+    if cross_compiler_prefix:
+        state = {'cross_compiler_prefix': cross_compiler_prefix}
```

Comment:
Maybe we could put this before the "update_state..." line above, and avoid a second state write, but it will work this way too.